parsec/core/mountpoint/manager.py

```diff
@@ -43,28 +49,71 @@ def get_mountpoint_runner():
             return fuse_mountpoint_runner
 
 
-def get_default_mountpoint(device_id: DeviceID):
-    if os.name == "nt":
-        return Path("Z:")
-    else:
-        return Path(f"/media/{device_id}")
+async def disabled_runner(*args, **kwargs):
+    raise MountpointDisabled()
+
+
+class MountpointManager:
+    def __init__(self, base_mountpoint, fs, runner, nursery):
+        self._base_mountpoint = base_mountpoint
+        self._fs = fs
+        self._runner = runner
+        self._nursery = nursery
+        self._mountpoint_tasks = {}
+
+    async def mount_workspace(self, workspace: str):
+        if workspace in self._mountpoint_tasks:
+            raise MountpointAlreadyMounted(f"Workspace `{workspace}` already mounted.")
+
+        try:
+            await self._fs.stat(f"/{workspace}")
+        except FileNotFoundError as exc:
+            raise MountpointConfigurationError(f"Workspace `{workspace}` doesn't exist") from exc
+
+        mountpoint = self._base_mountpoint / f"{self._fs.device.user_id}-{workspace}"
+        runner_task = await start_task(self._nursery, self._runner, workspace, mountpoint)
+        self._mountpoint_tasks[workspace] = runner_task
+
+    async def unmount_workspace(self, workspace: str):
+        if workspace not in self._mountpoint_tasks:
+            raise MountpointNotMounted(f"Workspace `{workspace}` not mounted.")
+
+        await self._mountpoint_tasks[workspace].cancel_and_join()
+        del self._mountpoint_tasks[workspace]
+
+    async def mount_all(self):
+        stat = await self._fs.stat("/")
+        for workspace in stat["children"]:
+            try:
+                await self.mount_workspace(workspace)
+            except MountpointAlreadyMounted:
+                pass
+
+    async def unmount_all(self):
+        for workspace in list(self._mountpoint_tasks.keys()):
+            await self.unmount_workspace(workspace)
 
 
 @asynccontextmanager
-async def mountpoint_manager(fs, event_bus, mountpoint, nursery, *, debug: bool = False, **config):
+async def mountpoint_manager_factory(
+    fs, event_bus, base_mountpoint, *, enabled=True, debug: bool = False, **config
+):
     config["debug"] = debug
 
-    # No mountpoint
-    if mountpoint is None:
-        yield None
-        return
+    if not enabled:
+        runner = disabled_runner
+    else:
+        runner = get_mountpoint_runner()
+        if not runner:
+            raise RuntimeError("Mountpoint support not available.")
+    curried_runner = partial(runner, config=config, fs=fs, event_bus=event_bus)
+
+    async with trio.open_nursery() as nursery:
```

Comment:
So the mountpoint manager doesn't use an external nursery anymore (`monitor_nursery` from logged core)? What was the problem?

Reply:
Separation of concerns !
Better avoid mixin nurseries (and use different nursery for different things)

Reply:
Agreed! This was bothering me while I was working on #145